app/controllers/spree/payments_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Spree
+  class PaymentsController < Spree::StoreController
+    ssl_required :redirect_to_authorize
+
+    respond_to :html
+
+    prepend_before_action :require_logged_in, only: :redirect_to_authorize
+
+    def redirect_to_authorize
+      @payment = Spree::Payment.find(params[:id])
+      authorize! :show, @payment
+
+      if url = @payment.cvv_response_message
+        redirect_to url
```

Comment:
we should whitelist the url or restrict it somehow, right?

Reply:
I added 9193cbcf9727777af890b4f08b5279943fb706c7 - it requires that the url be over https and contain "stripe.com". The Stripe API docs don't specify the format or where the redirect will go, but I think those are both safe assumptions.